File: backend/internal/service/time_convert_service.go
package service

import "time"

// TimeConvertService 时间转换服务
type TimeConvertService struct {
	// 可以添加缓存等优化
}

// NewTimeConvertService 创建时间转换服务
func NewTimeConvertService() *TimeConvertService {
	return &TimeConvertService{}
}

// Solar2Lunar 公历转农历
func (s *TimeConvertService) Solar2Lunar(year, month, day int) (*ConversionResult, error) {
	return Solar2Lunar(year, month, day)
}

// Lunar2Solar 农历转公历
func (s *TimeConvertService) Lunar2Solar(year, month, day int, isLeapMonth bool) (*ConversionResult, error) {
	return Lunar2Solar(year, month, day, isLeapMonth)
}

// GetDailyInfo 获取指定日期的详细信息（包含农历、节气、节日等）
func (s *TimeConvertService) GetDailyInfo(date time.Time) (*ConversionResult, error) {
	return Solar2Lunar(date.Year(), int(date.Month()), date.Day())
}

// GetSolarTerm 获取指定日期的节气
func (s *TimeConvertService) GetSolarTerm(date time.Time) (string, error) {
	res, err := s.GetDailyInfo(date)
	if err != nil {
		return "", err
	}
	return res.Term, nil
}

// GetFestivals 获取指定日期的节日列表（公历和农历）
func (s *TimeConvertService) GetFestivals(date time.Time) ([]string, error) {
	res, err := s.GetDailyInfo(date)
	if err != nil {
		return nil, err
	}
	var festivals []string
	if res.Festival != "" {
		festivals = append(festivals, res.Festival)
	}
	if res.LunarFestival != "" {
		festivals = append(festivals, res.LunarFestival)
	}
	return festivals, nil
}

File: backend-python/app/services/time_convert_service.py
"""时间转换服务"""

from datetime import datetime
from typing import Dict, Any
from app.utils.calendar import CalendarConverter


class TimeConvertService:
    """时间转换服务"""
    
    def __init__(self):
        self.converter = CalendarConverter()
    
    def solar_to_lunar(self, year: int, month: int, day: int) -> Dict[str, Any]:
        """公历转农历"""
        return self.converter.solar_to_lunar(year, month, day)
    
    def lunar_to_solar(self, year: int, month: int, day: int, is_leap_month: bool = False) -> Dict[str, Any]:
        """农历转公历"""
        return self.converter.lunar_to_solar(year, month, day, is_leap_month)
    
    def get_daily_info(self, date: datetime) -> Dict[str, Any]:
        """获取指定日期的详细信息（包含农历、节气、节日等）"""
        return self.solar_to_lunar(date.year, date.month, date.day)
    
    def get_solar_term(self, date: datetime) -> str:
        """获取指定日期的节气"""
        info = self.get_daily_info(date)
        return info.get("term", "")
    
    def get_festivals(self, date: datetime) -> list:
        """获取指定日期的节日列表（公历和农历）"""
        info = self.get_daily_info(date)
        festivals = []
        
        if info.get("festival"):
            festivals.append(info["festival"])
        
        if info.get("lunar_festival"):
            festivals.append(info["lunar_festival"])
        
        return festivals
